### PR TITLE
Plans Grid Refactor: restructure the plan type selector to be more maintainable

### PR DESCRIPTION
**Pull request by ddc22**
- Related to Automattic/growth-foundations#298

## Proposed Changes

* The above mentioned issue plans to add a new `kind` of toggle to the plan selector.
* The plan type selector code is currently unstructured and difficulty to reason about.
* This change moves the code into a more index.tsx, components, hooks structure.
* Completely removes Customer toggle type which is no longer required and which was introduced in, #27125

## Testing Instructions

* Should not materially change any behaviour of the grid interval selection should work well in all contexts.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plans-grid/components/plan-type-selector/components/customer-type-toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+type CustomerTypeProps = Pick< PlanTypeSelectorProps, 'customerType' | 'isInSignup' | 'title' >;
+
+/**
+ * Not sure where this is exactly used
+ * First brought to a standalone component in: https://github.com/Automattic/wp-calypso/pull/46466
+ * First introduction in https://github.com/Automattic/wp-calypso/pull/27125
+ *
+ * Probably can be removed.
+ */
+export const CustomerTypeToggle: React.FunctionComponent< CustomerTypeProps > = ( props ) => {
```
Comment (ddc22) [inline client/my-sites/plans-grid/components/plan-type-selector/components/customer-type-toggle.tsx]: Hi @adamziel @dzver Re: CustomerTypeToggle
This feature was first introduced in https://github.com/Automattic/wp-calypso/pull/27125
image

Do you know anyone that I can reach out to understand if this is being used anywhere ? If not we can probably clean this up permanently!

CC - @chriskmnds @southp

Comment (dzver) [reply client/my-sites/plans-grid/components/plan-type-selector/components/customer-type-toggle.tsx]: It shouldn't be in use, this was part of the attempts to have tiers of Personal and Business sites with several plans each.

**Comment by ddc22** [comment]
@chriskmnds appreciate your eyes on this since you worked on this part recently :)
I will be working on Automattic/growth-foundations#298 built on this foundation.

There was a merge conflict and this is how I resolved it ...
https://github.com/Automattic/wp-calypso/pull/84777/commits/953754e52b752785a1244e5ecf76d9cde54ae828

**Comment by ddc22** [comment]
Will merge this for now and pick up any rework in followup PRs.

**Pull request merged**